src/main/java/org/stellar/sdk/responses/SubmitTransactionResponse.java

```diff
@@ -97,15 +100,25 @@ public Long getOfferIdFromResult(int position) {
             return null;
         }
 
-        if (result.getResult().getResults()[position].getTr().getDiscriminant() != OperationType.MANAGE_OFFER) {
-            return null;
+        OperationResult operationResult = result.getResult().getResults()[position];
+        OperationType operationType = operationResult.getTr().getDiscriminant();
+        OperationResult.OperationResultTr operationResultTr = operationResult.getTr();
+
+        if (operationType == OperationType.MANAGE_SELL_OFFER) {
+            if (operationResultTr.getManageSellOfferResult().getSuccess().getOffer().getOffer() == null) {
+                return null;
+            }
+            return operationResultTr.getManageSellOfferResult().getSuccess().getOffer().getOffer().getOfferID().getInt64();
         }
 
-        if (result.getResult().getResults()[0].getTr().getManageOfferResult().getSuccess().getOffer().getOffer() == null) {
```

Comment:
I think this is incorrect, it should be `if (result.getResult().getResults()[position].getTr().getManageOfferResult().getSuccess().getOffer().getOffer() == null) {`. If you agree with me, you can cherry-pick 555ffcd8d02f9a950564210435a45710c3163a61 first.

Reply:
Yes, you are correct. Looks like a bug introduced when merging etc.